include/slb/type_traits.hpp

```diff
@@ -1023,10 +1023,28 @@ using std::conjunction;
 using std::disjunction;
 using std::negation;
 #else
-// TODO
-// template <typename... Bs> struct conjunction;
-// template <typename... Bs> struct disjunction;
-// template <typename B> struct negation;
+template <typename... Bs>
+struct conjunction : slb::true_type {};
+
+template <typename B>
+struct conjunction<B> : B {};
+
+template <typename B, typename... Bs>
+struct conjunction<B, Bs...>
+    : slb::conditional_t<static_cast<bool>(B::value), conjunction<Bs...>, B> {};
```

Comment:
The wording uses `bool(B::value)`, which is a C-style cast, rather than `static_cast<bool>(B::value)`. I tried very hard to come up with a situation in which they would be different, but found none. My preference would be to stick with `static_cast` unless and until we find a divergence.

Interestingly, `conjunction`/`disjunction` require `B::value` to also be (implicitly) convertible to `bool`, whereas `negation` doesn't.